lib.php

```diff
@@ -474,16 +474,33 @@ function block_coursefeedback_get_implemented_languages($feedbackid, $questionid
 }
 
 /**
- * Computes the next free questionid, is also used to detect the amount of questions the FB has.
+ * Computes the next available (unused) questionid for a given feedback.
  *
- * @param int $feedbackid
- * @return int - Next availble question id number.
+ * @param int $feedbackid The feedback ID.
+ * @return int The next available question ID.
+ * @throws dml_exception If the database query fails.
  */
-function block_coursefeedback_get_questionid($feedbackid) {
+function block_coursefeedback_get_questionid(int $feedbackid): int {
     global $DB;
-    $feedbackid = intval($feedbackid);
-    $n = $DB->get_field("block_coursefeedback_questns", "MAX(questionid)", array("coursefeedbackid" => $feedbackid));
-    return $n ? ($n + 1) : 1;
+    $sql = "SELECT COALESCE(MAX(questionid), 0) + 1 
+              FROM {block_coursefeedback_questns} 
+             WHERE coursefeedbackid = :feedbackid";
+    return $DB->get_field_sql($sql, ["feedbackid" => $feedbackid]);
+}
+
+/**
+ * Calculates how many questions a feedback has.
+ *
+ * @param int $feedbackid The feedback ID.
+ * @return int How much Questions for this Feedback.
+ * @throws dml_exception If the database query fails.
+ */
+function block_coursefeedback_get_questioncount(int $feedbackid): int {
+    global $DB;
+    $sql = "SELECT COUNT(DISTINCT questionid) 
+              FROM {block_coursefeedback_questns} 
+             WHERE coursefeedbackid = :feedbackid";
+    return (int) $DB->get_field_sql($sql, ["feedbackid" => $feedbackid]);
```

Comment:
That cast is redundant because you have a return type annotation for the function.
```php
function f(): int { return '1'; }
var_dump(f());
```
Prints `int(1)`.

```suggestion
    return $DB->get_field_sql($sql, ["feedbackid" => $feedbackid]);
```